### PR TITLE
Cast sparse placeholder shape to int64 to avoid tensorflow bug

### PR DESCRIPTION
**Pull request by michaelosthege**
casting the shape to int64 resolves issue [5225](https://github.com/fchollet/keras/issues/5225) with a workaround to avoid tensorflow issue [6749](https://github.com/tensorflow/tensorflow/issues/6749).

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ def placeholder(shape=None, ndim=None, dtype=None, sparse=False, name=None):
         if ndim:
             shape = tuple([None for _ in range(ndim)])
     if sparse:
-        x = tf.sparse_placeholder(dtype, shape=shape, name=name)
+        x = tf.sparse_placeholder(dtype, shape=np.array(shape).astype("int64"), name=name)
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Please use `'` as quote character for consistency.

Comment (michaelosthege) [inline keras/backend/tensorflow_backend.py]: done.
unfortunately there's another problem that arises:

```python
/home/ubuntu/miniconda3/envs/venv35/lib/python3.5/site-packages/keras/backend/tensorflow_backend.py in __call__(self, inputs)
   2073         session = get_session()
   2074         updated = session.run(self.outputs + [self.updates_op],
-> 2075                               feed_dict=feed_dict)
   2076         return updated[:len(self.outputs)]
   2077 

/home/ubuntu/miniconda3/envs/venv35/lib/python3.5/site-packages/tensorflow/python/client/session.py in run(self, fetches, feed_dict, options, run_metadata)
    765     try:
    766       result = self._run(None, fetches, feed_dict, options_ptr,
--> 767                          run_metadata_ptr)
    768       if run_metadata:
    769         proto_data = tf_session.TF_GetBuffer(run_metadata_ptr)

/home/ubuntu/miniconda3/envs/venv35/lib/python3.5/site-packages/tensorflow/python/client/session.py in _run(self, handle, fetches, feed_dict, options, run_metadata)
    944                 % (np_val.shape, subfeed_t.name, str(subfeed_t.get_shape())))
    945           if not self.graph.is_feedable(subfeed_t):
--> 946             raise ValueError('Tensor %s may not be fed.' % subfeed_t)
    947           subfeed_name = compat.as_bytes(subfeed_t.name)
    948           feed_dict_string[subfeed_name] = np_val

ValueError: Tensor Tensor("Input_2/shape_1:0", shape=(2,), dtype=int64) may not be fed.
```

Looks like the int64 shape Tensor can not be fed unless that bug is resolved in tensorflow.
Previously I was able to cheat my way around this issue by reshaping everything that got matmul-ed with the sparse tensor (ending up with None shapes) back into the correct shape.

In the meantime from my understanding it should be quite impossible to feed any sparse placeholder __at all__ (with or without the proposed change).

**Comment by michaelosthege** [comment]
This doesn't make any sense.. The casting workaround doesn't work because tensorflow won't accept to be fed with an int64 Tensor either. (see my [comment here](https://github.com/tensorflow/tensorflow/issues/6749#issuecomment-286632439))

Unless I am missing something, the only way to keep it in a working state is to remove the shape argument until the tensorflow issue is resolved.

**Comment by rolfe** [comment]
I also seem to be having the same problem.  What is the status of this bug?

**Comment by michaelosthege** [comment]
The tf issue is still unresolved.
I am actually surprised that there aren't that many people complaining about it.

My workaround is to edit tensorflow_backend.py and remove the shape=shape from sparse placeholder creation.

**Comment by leriomaggio** [comment]
Any updates on this!
Indeed the actual code of Keras is not able to apply your workaround - so you cannot pass shape as a `numpy.ndarray` as input as it crashes on the `if not shape` condition...

**Comment by yaseenit** [comment]
Any updates on this please

**Pull request closed without merging**